### ISSUE TITLE
Add decorators to all TMC modules

### ISSUE THREAD

**Issue by gogonzo** [insightsengineering/teal#1371]
Blocked by https://github.com/insightsengineering/teal/issues/1368

- Assign yourself to a single module by adding your name next to the item. 
- add the decorator functionality to the module and update relevant documentation by including the decorator
- Keep consistent output names - `table`, `listing`, `graph`
- PR link should be added next to the item
- once merged ✔️ the item

Items:

### Part 1

- [x] tm_a_gee @llrs-roche  @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1262
- [x] tm_g_ci @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1265
- [x] tm_g_forest_rsp @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1266
- [x] tm_g_forest_tte @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1264
- [x] tm_g_ipp @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1263
- [x] tm_g_km @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1254
- [x] tm_g_lineplot @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1255
- [x] tm_g_pp_patient_timeline @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1270
- [x] tm_g_pp_therapy @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1268
- [x] tm_g_pp_vitals @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1261
- [x] tm_t_abnormality @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1257
- [x] tm_t_abnormality_by_worst_grade @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1260
- [x] tm_t_ancova @m7pr https://github.com/insightsengineering/teal.modules.clinical/pull/1256
- [x] tm_g_barchart_simple @m7pr @averissimo  https://github.com/insightsengineering/teal.modules.clinical/pull/1267
- [x] tm_g_pp_adverse_events @m7pr @averissimo  https://github.com/insightsengineering/teal.modules.clinical/pull/1269

#### Moved from part 2 to part 1
- [x] tm_t_binary_outcome @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1273
- [x] tm_t_coxreg @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1274
- [x] tm_t_events @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1275
- [x] tm_t_events_by_grade @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1276
- [x] tm_t_events_patyear @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1277
- [x] tm_t_events_summary @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1278
- [x] tm_t_exposure @llrs-roche https://github.com/insightsengineering/teal.modules.clinical/pull/1279
- [x] tm_t_tte @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1280
- [x] tm_t_mult_events @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1281
- [x] tm_t_pp_basic_info @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1282

### Part 2
- [x] tm_t_logistic @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1290
- [x] tm_t_pp_laboratory @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1291
- [x] tm_t_pp_medical_history @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1289
- [x] tm_t_pp_prior_medication @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1288
- [x] tm_t_shift_by_arm @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1287
- [x] tm_t_shift_by_arm_by_worst @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1286
- [x] tm_t_shift_by_grade @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1285
- [x] tm_t_smq @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1292
- [x] tm_t_summary @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1293
- [x] tm_t_summary_by @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1294

#### Moved from part 1
- [x] tm_a_mmrm **this one is crazy** @averissimo https://github.com/insightsengineering/teal.modules.clinical/pull/1300

**Comment by averissimo** [comment]
The modules have some outputs that are not included in the reporter, or if they are, they have no styling associated (as they are passed to reporter as an R `data.frame`).

One of the cases is `tm_g_barchart_simple` (see image at bottom)

For these I believe we have 2 choices:

1. Decorate the main output in cases where the table is displayed via UI element
    - Such as `{rtables}`, `{DataTable}`, ... objects
    - Not for when it's being displayed from a `data.frame` or other data-only objects
2. Do not decorate!
    - In other words, only decorate objects that have styling:
        -  `{rtables}` (`rtables`/`TableTree`/`ElementaryTable`) & `{rlistings}` objects (`listing_df`)
        - `ggplot`
        - `grob`
        - ...

After talking with the team and continuing the discussion with @llrs-roche we think that for now, we should go with `2.` 

Option `1.` is a new feature request.

----

#### `tm_g_barchart_simple` screenshot of module & report card

![image](https://github.com/user-attachments/assets/c4578322-5ad7-4fc6-bbb3-2d07b13aa2d0)

**Comment by averissimo** [comment]
There are 3 types of outputs in TMC and so far we have been allowing decorating on all 3 types (as well as the module allow).

1. Shown \*\***BOTH**\*\* in module tab and in reporter card
    - `ggplot2`/`grob` and `rtables`
2. Same as `1.`, but the data passed to reporter is raw data
    - Simple visualizations of `data.frame`
    - Interactive visualizations that are displayed with [DataTable](https://datatables.net/), but passed to reporter as data-only
3. Shown \*\***ONLY**\*\* in module tab _(see image in bottom for example)_
    - Such as tables/DataTable for selected datapoints in plot (brushing)
    - Some visualization-only data _(may not even have reporter card)_

ℹ️ After talking with the team and continuing the discussion with @llrs-roche we think that the approach is out of scope, we should move only to decorate `1.`, such as:

- `ggplot`
- `grob`
- `{rtables}` (`rtables`/`TableTree`/`ElementaryTable`) & `{rlistings}` objects (`listing_df`)
    - Types accepted by reporter
- _(other cases that may appear)_

----

#### `tm_g_barchart_simple` screenshot of module & report card

![image](https://github.com/user-attachments/assets/9f254c33-ad53-4b7a-9bbd-b4538ec1492f)
